compiler/bindings/python/test/api/tuner_api_test.py

```diff
@@ -336,3 +336,242 @@ def test_igemm_conv_details():
+    assert list(dims.k) == [2], f"Expected k=[2], got {list(dims.k)}"
+    assert list(dims.kB) == [3], f"Expected kB=[3], got {list(dims.kB)}"
+    assert list(dims.batch) == [], f"Expected no batch dims, got {list(dims.batch)}"
+
+
+@run
+def test_infer_scaled_contraction_dimensions():
+    # Test 1: Verify dimension inference on a scaled matmul operation.
+    module_str = """
+        module {
+            func.func @scaled_matmul(%lhs: tensor<16x4x32xf4E2M1FN>, %rhs: tensor<16x4x32xf4E2M1FN>,
+                                     %lhs_scales: tensor<16x4xf8E8M0FNU>, %rhs_scales: tensor<16x4xf8E8M0FNU>,
+                                     %out: tensor<16x16xf32>) -> tensor<16x16xf32> {
+                %result = linalg.generic {
+                    indexing_maps = [
+                        affine_map<(d0, d1, d2, d3) -> (d0, d2, d3)>,
+                        affine_map<(d0, d1, d2, d3) -> (d1, d2, d3)>,
+                        affine_map<(d0, d1, d2, d3) -> (d0, d2)>,
+                        affine_map<(d0, d1, d2, d3) -> (d1, d2)>,
+                        affine_map<(d0, d1, d2, d3) -> (d0, d1)>
+                    ],
+                    iterator_types = ["parallel", "parallel", "reduction", "reduction"],
+                    root_op
+                } ins(%lhs, %rhs, %lhs_scales, %rhs_scales : tensor<16x4x32xf4E2M1FN>, tensor<16x4x32xf4E2M1FN>, tensor<16x4xf8E8M0FNU>, tensor<16x4xf8E8M0FNU>)
+                  outs(%out : tensor<16x16xf32>) {
+                ^bb0(%a: f4E2M1FN, %b: f4E2M1FN, %a_scale: f8E8M0FNU, %b_scale: f8E8M0FNU, %acc: f32):
+                    %a_scaled = arith.scaling_extf %a, %a_scale : f4E2M1FN, f8E8M0FNU to f32
+                    %b_scaled = arith.scaling_extf %b, %b_scale : f4E2M1FN, f8E8M0FNU to f32
+                    %prod = arith.mulf %a_scaled, %b_scaled : f32
+                    %sum = arith.addf %acc, %prod : f32
+                    linalg.yield %sum : f32
+                } -> tensor<16x16xf32>
+                return %result : tensor<16x16xf32>
+            }
+        }
+    """
+    input_module = ir.Module.parse(module_str)
+    root_op_list = iree_codegen.get_tuner_root_ops(input_module)
+    assert len(root_op_list) == 1, "Should have exactly one root op"
+    scaled_op = root_op_list[0]
+
+    # Verify it's a scaled contraction first.
+    assert iree_codegen.isa_scaled_contraction_op(
+        scaled_op
+    ), "Operation should be recognized as scaled contraction"
+
+    # Test dimension inference.
+    dims = iree_codegen.infer_scaled_contraction_dimensions(scaled_op)
+    assert dims is not None, "Should successfully infer dimensions"
+
+    # Verify the inferred dimensions match expected values.
+    # For the given indexing maps:
+    # d0 = M (parallel) -> m,
+    # d1 = N (parallel) -> n,
+    # d2 = Ko (reduction) -> k,
+    # d3 = Kb (reduction, block dim) -> kB.
+    assert list(dims.m) == [0], f"Expected m=[0], got {list(dims.m)}"
```

Comment:
```suggestion
    assert dims.m == [0], f"Got {list(dims.m)}"
```
The expected values are already visible in the assertion itself.

So apply this to all the tests.